stan/math/rev/core/arena_matrix.hpp

```diff
@@ -110,23 +154,32 @@ class arena_matrix : public Eigen::Map<MatrixType> {
    * @param a expression to evaluate into this
    * @return `*this`
    */
-  template <typename T>
-  arena_matrix& operator=(const T& a) {
-    // do we need to transpose?
-    if ((RowsAtCompileTime == 1 && T::ColsAtCompileTime == 1)
-        || (ColsAtCompileTime == 1 && T::RowsAtCompileTime == 1)) {
-      // placement new changes what data map points to - there is no allocation
-      new (this) Base(
-          ChainableStack::instance_->memalloc_.alloc_array<Scalar>(a.size()),
-          a.cols(), a.rows());
-
+  template <typename T, require_not_arena_matrix_t<T>* = nullptr>
+  arena_matrix& operator=(T&& a) {
+    using T_t = std::decay_t<T>;
+    if (std::is_rvalue_reference<T&&>::value && is_plain_type<T_t>::value) {
+      // Note: plain_type_t here does nothing since T_t is plain type
+      auto other = make_chainable_ptr(plain_type_t<MatrixType>(std::move(a)));
```

Comment:
If the `plain_type_t` is needed for some reason even though it does nothing, can you change the comment to explain why? Otherwise better to just remove the `plain_type_t`